tests/functions/test_join_conditional.py

```diff
@@ -0,0 +1,26 @@
+from janitor import join_conditional
+import pandas as pd
+import numpy as np
+import pytest
+
+
+@pytest.mark.functions
+def test_join_conditional():
```

Comment:
Could we get a docstring here, while also including in there that this is an example-based test?

Reply:
**Action item:** As per title.